File: A1/A1-011/A1-011.cpp
#include<iostream>
using namespace std;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);

    // Input
    string text;
    cin >> text;
    // Output
    char letter = text[0];
    int cnt = 1;
    for(int i = 1; i < text.size(); i++)
    {
        // Count if it is the same letter
        if(text[i] == letter) { cnt++; }
        // Otherwise, output and change to the new letter
        else
        {
            cout << cnt << letter;
            letter = text[i];
            cnt = 1;
        }
    }
    // Don't forget to output the last one
    cout << cnt << letter;
    return 0;
}

File: A1/A1-011/A1-011.py
# --------------------------------------------------
# File Name : A1-011.py
# Problem   : THEOS
# Author    : Worralop Srichainont
# Date      : 2026-02-08
# --------------------------------------------------

# Input text
text = input().strip()

# Initialize current letter and counts
curr_letter = text[0]
curr_count = 1

# Iterate each letter and count
for i in range(1, len(text)):
    # If the current letter is the same, just count
    if text[i] == curr_letter:
        curr_count += 1

    # Otherwise, output and change to the new letter
    else:
        # Output the current count
        print(f"{curr_count}{curr_letter}", end="")

        # Change to the new letter
        curr_letter = text[i]
        curr_count = 1

# Do not forget to output the last one
print(f"{curr_count}{curr_letter}")
